Traceback (most recent call last):
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/mhjensen/miniforge3/envs/myenv/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
%matplotlib inline

# Common imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from IPython.display import display
import os

# Where to save the figures and data files
PROJECT_ROOT_DIR = "Results"
FIGURE_ID = "Results/FigureFiles"
DATA_ID = "DataFiles/"

if not os.path.exists(PROJECT_ROOT_DIR):
    os.mkdir(PROJECT_ROOT_DIR)

if not os.path.exists(FIGURE_ID):
    os.makedirs(FIGURE_ID)

if not os.path.exists(DATA_ID):
    os.makedirs(DATA_ID)

def image_path(fig_id):
    return os.path.join(FIGURE_ID, fig_id)

def data_path(dat_id):
    return os.path.join(DATA_ID, dat_id)

def save_fig(fig_id):
    plt.savefig(image_path(fig_id) + ".png", format='png')

infile = open(data_path("MassEval2016.dat"),'r')


# Read the experimental data with Pandas
Masses = pd.read_fwf(infile, usecols=(2,3,4,6,11),
              names=('N', 'Z', 'A', 'Element', 'Ebinding'),
              widths=(1,3,5,5,5,1,3,4,1,13,11,11,9,1,2,11,9,1,3,1,12,11,1),
              header=39,
              index_col=False)

# Extrapolated values are indicated by '#' in place of the decimal place, so
# the Ebinding column won't be numeric. Coerce to float and drop these entries.
Masses['Ebinding'] = pd.to_numeric(Masses['Ebinding'], errors='coerce')
Masses = Masses.dropna()
# Convert from keV to MeV.
Masses['Ebinding'] /= 1000

# Group the DataFrame by nucleon number, A.
Masses = Masses.groupby('A')
# Find the rows of the grouped DataFrame with the maximum binding energy.
Masses = Masses.apply(lambda t: t[t.Ebinding==t.Ebinding.max()])
A = Masses['A']
Z = Masses['Z']
N = Masses['N']
Element = Masses['Element']
Energies = Masses['Ebinding']

# Now we set up the design matrix X
X = np.zeros((len(A),5))
X[:,0] = 1
X[:,1] = A
X[:,2] = A**(2.0/3.0)
X[:,3] = A**(-1.0/3.0)
X[:,4] = A**(-1.0)
# Then nice printout using pandas
DesignMatrix = pd.DataFrame(X)
DesignMatrix.index = A
DesignMatrix.columns = ['1', 'A', 'A^(2/3)', 'A^(-1/3)', '1/A']
display(DesignMatrix)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mFileNotFoundError[0m                         Traceback (most recent call last)
Input [0;32mIn [1][0m, in [0;36m<cell line: 33>[0;34m()[0m
[1;32m     30[0m [38;5;28;01mdef[39;00m [38;5;21msave_fig[39m(fig_id):
[1;32m     31[0m     plt[38;5;241m.[39msavefig(image_path(fig_id) [38;5;241m+[39m [38;5;124m"[39m[38;5;124m.png[39m[38;5;124m"[39m, [38;5;28mformat[39m[38;5;241m=[39m[38;5;124m'[39m[38;5;124mpng[39m[38;5;124m'[39m)
[0;32m---> 33[0m infile [38;5;241m=[39m [38;5;28;43mopen[39;49m[43m([49m[43mdata_path[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43mMassEval2016.dat[39;49m[38;5;124;43m"[39;49m[43m)[49m[43m,[49m[38;5;124;43m'[39;49m[38;5;124;43mr[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m     36[0m [38;5;66;03m# Read the experimental data with Pandas[39;00m
[1;32m     37[0m Masses [38;5;241m=[39m pd[38;5;241m.[39mread_fwf(infile, usecols[38;5;241m=[39m([38;5;241m2[39m,[38;5;241m3[39m,[38;5;241m4[39m,[38;5;241m6[39m,[38;5;241m11[39m),
[1;32m     38[0m               names[38;5;241m=[39m([38;5;124m'[39m[38;5;124mN[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mZ[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mA[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mElement[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mEbinding[39m[38;5;124m'[39m),
[1;32m     39[0m               widths[38;5;241m=[39m([38;5;241m1[39m,[38;5;241m3[39m,[38;5;241m5[39m,[38;5;241m5[39m,[38;5;241m5[39m,[38;5;241m1[39m,[38;5;241m3[39m,[38;5;241m4[39m,[38;5;241m1[39m,[38;5;241m13[39m,[38;5;241m11[39m,[38;5;241m11[39m,[38;5;241m9[39m,[38;5;241m1[39m,[38;5;241m2[39m,[38;5;241m11[39m,[38;5;241m9[39m,[38;5;241m1[39m,[38;5;241m3[39m,[38;5;241m1[39m,[38;5;241m12[39m,[38;5;241m11[39m,[38;5;241m1[39m),
[1;32m     40[0m               header[38;5;241m=[39m[38;5;241m39[39m,
[1;32m     41[0m               index_col[38;5;241m=[39m[38;5;28;01mFalse[39;00m)

[0;31mFileNotFoundError[0m: [Errno 2] No such file or directory: 'DataFiles/MassEval2016.dat'
FileNotFoundError: [Errno 2] No such file or directory: 'DataFiles/MassEval2016.dat'

